marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/models/deleteModel.sjs

```diff
@@ -19,8 +19,22 @@ const ds = require("/data-hub/5/data-services/ds-utils.sjs");
 const entityLib = require("/data-hub/5/impl/entity-lib.sjs");
 
 var entityName;
-if (entityName === undefined || entityName === null) {
+if (!entityName) {
   ds.throwBadRequest("Must specify a name in order to delete an entity model");
 }
 
+const entityModel = entityLib.findModelByEntityName(entityName);
+if (!entityModel) {
+  ds.throwServerError(`Could not find entity model with name: ${entityName}`);
+}
+
+const entityTypeId = entityLib.getEntityTypeId(entityModel, entityName);
+const entityModelUri = entityLib.getModelUri(entityName);
+
+const steps = entityLib.findModelReferencesInSteps(entityName, entityTypeId);
+if (steps.length) {
+  ds.throwServerError(`Cannot delete the entity type '${entityName}' because it is referenced by the following step names: ${steps}`);
```

Comment:
I think we have to say "step and/or mapping names", since the list may contain legacy mapping names, which can be different from the names of the steps that reference them.